carbon-design-system/carbon
# feat(styles): add support for IBM Plex from the `@ibm/plex` package

This PR adds in support for bringing in IBM Plex to Carbon through `scss/fonts`. This update allows teams to bundle the latest version of `@ibm/plex` into their project, in particular for bundlers like webpack. It also allows teams that self-host to use our system to correctly determine `url()` values for `@font-face` blocks.

Overall, this PR by default allows IBM Plex Mono, Sans, Serif to be included with the light, regular, and semibold font weights. It also allows teams to include other font weights and other font families such as:

- IBM Plex Sans Arabic
- IBM Plex Sans Devanagari
- IBM Plex Sans Hebrew
- IBM Plex Sans Thai
- IBM Plex Sans Thai Looped

This PR **does not** include the following:

- IBM Plex Sans JP
- IBM Plex Sans KR

#### Changelog

**New**

- Add `scss/fonts` entrypoint to `@carbon/styles`

**Changed**

- Update how fonts are brought in with `@carbon/styles`
- Update how storybook brings in fonts in `carbon-react`

**Removed**

- Removed `scss/_font-face.scss`

#### Testing / Reviewing

- `packages/styles/scss/fonts`
  - Review the implementation and see if it makes sense to you
  - Make sure implementation is tested and the tests pass
  - Review the `scss/fonts/README.md` file and make content suggestions where appropriate if something is unclear or missing from the docs
- `packages/carbon-react`
  - Verify that all files from `packages/styles/scss/fonts` are correctly re-exported in `packages/carbon-react/scss/fonts`
  - In the storybook, visit the Elements / IBM Plex story and do the following:
    - Verify each font loads as a network resource
    - Verify fonts with unicode ranges only load the relevant unicode range
    - There's a video below of what this would look like

https://user-images.githubusercontent.com/3901764/144306731-2327256c-2f97-40ca-a34d-8096b84a9032.mov

At the end, you'll see that I noticed that IBM Plex Sans Thai was loaded incorrectly and didn't not have the correct font resource loaded
